src/command/restore/restore.c

```diff
@@ -1329,6 +1334,26 @@ restoreRecoveryOption(unsigned int pgVersion)
             strLstSort(recoveryOptionKey, sortOrderAsc);
         }
 
+        // If archive-mode is not preserve
+        const String *archiveMode = cfgOptionStr(cfgOptArchiveMode);
+
+        if (!strEq(archiveMode, ARCHIVE_MODE_PRESERVE_STR))
+        {
+            if (pgVersion < PG_VERSION_12)
+            {
+                THROW_FMT(
+                    OptionInvalidError,
+                    "option '" CFGOPT_ARCHIVE_MODE "' is not supported on " PG_NAME " < " PG_VERSION_12_STR "\n"
+                        "HINT: archive_mode should be set manually in postgresql.conf.");
```

Comment:
I would have this explicitly say "archive_mode should be set to off manually in postgresql.conf" or something along those lines, since that's what the user asked to try and do.

Reply:
Sounds good, done that way.